### ISSUE TITLE
ExtUtils::MakeMaker calls by default /bin/tar which creates problematic PAX archives

### ISSUE THREAD

**Issue by pali** [Perl-Toolchain-Gang/ExtUtils-MakeMaker#349]
ExtUtils::MakeMaker uses by default system /bin/tar binary for creating distribution archives when invoking `make dist`. On more systems /bin/tar binary by default creates PAX archives instead of classic TAR archives. When you try to unpack PAX archive by a program which understands only ustar (and not pax), it creates PaxHeader subdirectory and in this subdirectory creates random files and their content has raw PAX extended attributes. So it is common that such tar program creates PaxHeader/Makefile.PL with some random garbage. It is not valid Perl code and existing it fails. So on such systems it is not possible to install perl distributions from PAX archives. There are more bug reports about this issue, e.g.:
https://github.com/Dual-Life/Devel-PPPort/issues/183
https://huntingbears.nl/2015/02/17/new-tar-paxheaders-and-installing-from-cpan/
http://www.perlmonks.org/?node_id=1004571

So correct way to generate a new distribution is to use ustar archive and not PAX archive.

Perl since 5.10 has in Core module Archive::Tar and `ptar` binary which creates TAR archives, not PAX. So ExtUtils::MakeMaker could switch from /bin/tar to ptar by default to fix this problem and prevent creating of PAX archives on systems which /bin/tar binary by default creates PAX and not ustar. IIRC new versions of GNU tar already switched default to PAX, so Perl core `ptar` looks like a solution for this problem.

There is already proposed change to use `--format=ustar` flag for /bin/tar, but in discussion was a point that this option works correctly only for GNU tar implementation. For BSD tar implementation it in some cases creates empty tar archives... See discussion: https://github.com/xsawyerx/module-starter/pull/73

**Comment by johannessen** [comment]
`ptar` does look like a solution here. However, a change to use it by default may break the Makefile.PL of users who tried to work around this issue by adding `--format=ustar` to their `TARFLAGS`.

Perhaps Archive::Tar’s `ptar` should be changed so that it accepts (and ignores) an argument `--format=ustar`?

Or ExtUtils::MakeMaker could remove `--format=ustar` from `TARFLAGS` when `TAR` isn’t defined (and defaults to `ptar`)?

**Comment by Leont** [comment]
> Perhaps Archive::Tar’s ptar should be changed so that it accepts (and ignores) an argument --format=ustar?

Sounds reasonable regardless of what we do with MakeMaker. https://github.com/jib/archive-tar-new

> Or ExtUtils::MakeMaker could remove --format=ustar from TARFLAGS when TAR isn’t defined (and defaults to ptar)?

That would require detecting exactly what tar we're working with here, and only add it for tars where it's needed/supported.

**Comment by johannessen** [comment]
>> Or ExtUtils::MakeMaker could remove `--format=ustar` from `TARFLAGS` when `TAR` isn’t defined (and defaults to `ptar`)?
> 
> That would require detecting exactly what tar we're working with here, and only add it for tars where it's needed/supported.

I don’t think it would – we’d know we’d be working with `ptar`. But if https://github.com/jib/archive-tar-new/pull/45 is accepted, I suppose it doesn’t matter.

**Comment by choroba** [comment]
It's been accepted.

**Comment by 0-wiz-0** [comment]
I've filed a [bug report against Params::SomeUtil](https://github.com/uperl/Params-SomeUtil/issues/10) that was rejected, pointing to https://uperl.github.io/faq.html which lead me here.
Can ExtUtils::MakeMaker please made to at least avoid including macOS extended attributes in the generated tarballs by default, please?
